parsers/IN.py

```diff
@@ -161,6 +219,125 @@ def fetch_consumption(
     return data
 
 
+def fetch_npp_production(
+    zone_key: str,
+    target_datetime: datetime,
+    session: Session = Session(),
+    logger: Logger = getLogger(__name__),
+) -> dict:
+    """Gets production for conventional thermal, nuclear and hydro from NPP daily reports
+    This data most likely doesn't inlcude distributed generation"""
+    npp_url = "https://npp.gov.in/public-reports/cea/daily/dgr/{date:%d-%m-%Y}/dgr2-{date:%Y-%m-%d}.xls".format(
+        date=target_datetime
+    )
+    r: Response = session.get(npp_url)
+    if r.status_code == 200:
+        df_npp = pd.read_excel(r.content, header=3)
+        df_npp = df_npp.rename(
+            columns={
+                df_npp.columns[0]: "power_station",
+                df_npp.columns[2]: "production_mode",
+                "TODAY'S\nACTUAL\n": "value",
+            }
+        )
+        df_npp = df_npp[["power_station", "production_mode", "value"]]
+        df_npp = df_npp.iloc[1:].copy()
+        df_npp["production_mode"] = df_npp["production_mode"].ffill()
+
+        df_npp["region"] = (
+            df_npp["power_station"]
+            .apply(lambda x: NPP_REGION_MAPPING[x] if x in NPP_REGION_MAPPING else None)
+            .ffill()
+        )
+        df_zone = df_npp.loc[df_npp["region"] == zone_key].copy()
+        df_zone = df_zone.loc[~df_zone.power_station.isna()]
+        df_zone = df_zone[df_zone.power_station.str.contains("TYPE:")]
+        df_zone = df_zone[["production_mode", "value"]]
+        df_zone = df_zone.groupby(["production_mode"]).sum()
+        production = {}
+        for mode in df_zone.index:
+            production[NPP_MODE_MAPPING[mode]] = round(
+                df_zone.iloc[df_zone.index.get_indexer_for([mode])[0]].get("value")
+                / CONVERSION_MWH_MW,
+                3,
+            )
+        return production
+    else:
+        raise ParserException(
+            parser="IN.py",
+            message=f"{target_datetime}: {zone_key} conventional production data is not available",
```

Comment:
```suggestion
            message=f"{target_datetime}: {zone_key} conventional production data is not available : [{r.status_code}]",
```
I would suggest to log the status code, this will allow us a faster troubleshooting.